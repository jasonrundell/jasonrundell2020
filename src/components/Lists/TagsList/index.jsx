import React from 'react'
import PropTypes from 'prop-types'

import Tag from '../../Tag'

import styles from './TagsList.module.scss'

const TagsList = ({ tags, orderBy }) => {
  // TODO: Array sort
  return (
    <ul className={styles.root}>
      {tags.map(tag => (
        <li className={styles.listItem} key={tag.id}>
          <Tag text={tag.text} />
        </li>
      ))}
    </ul>
  )
}

TagsList.defaultProps = {
  orderBy: 'ASC',
}

TagsList.propTypes = {
  tags: PropTypes.array.isRequired,
  orderBy: PropTypes.oneOf(['ASC', 'DESC']),
}

export default TagsList
