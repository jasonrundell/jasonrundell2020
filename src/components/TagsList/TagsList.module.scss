.root {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  list-style: none;
  // justify-content: flex-start;
  // align-items: center;
  padding: 0;
}
.list-item {
  // flex: 0 1 25%;
}
