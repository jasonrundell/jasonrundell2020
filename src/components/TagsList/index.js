import React from 'react'
import PropTypes from 'prop-types'
import { Tag } from 'antd'

import styles from './TagsList.module.scss'

const TagsList = ({ tags = [] }) => (
  <ul className={styles.root}>
    {tags.map(tag => (
      <li className={styles.listItem} key={tag}>
        <Tag>{tag}</Tag>
      </li>
    ))}
  </ul>
)

TagsList.propTypes = {
  tags: PropTypes.array.isRequired,
}

export default TagsList
