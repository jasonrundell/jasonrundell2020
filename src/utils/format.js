/**
 *  Letter case conversion functions
 *  [Letter case](https://en.wikipedia.org/wiki/Letter_case#Special_case_styles)
 *
 *  Converts only letter case-separated or hyphen-delimited strings
 */

/**
 *  Converts hyphen-delimited `string` to camel case
 *
 *  @example
 *  'foo-bar' => 'fooBar'
 *  'foo--bar' => 'fooBar'
 *
 *  @param {string} string
 */
function toCamelCase(string) {
  return string.replace(/(-+)([a-z])/g, ($0, $1, $2) => $2.toUpperCase())
}

/**
 *  Converts hyphen-delimited `string` to pascal case (aka upper camel case)
 *
 *  @example
 *  'foo-bar' => 'FooBar'
 *  'foo--bar' => 'FooBar'
 *
 *  @param {string} string
 */
function toPascalCase(string) {
  return toCamelCase(string.replace(/^[a-z]/, $0 => $0.toUpperCase()))
}

export { toCamelCase, toPascalCase }
