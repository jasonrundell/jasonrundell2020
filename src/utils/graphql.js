export function edgesToObject(edgesData) {
  const nodes = edgesData.map(item => {
    return item.node
  })
  const finalObject = nodes.map(item => {
    return item
  })

  return finalObject
}
