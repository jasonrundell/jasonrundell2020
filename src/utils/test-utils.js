export function catchWarnings() {
  const originalConsoleError = global.console.error
  beforeEach(() => {
    /**
     * I Want to know if values passed to props fail the type check
     * */
    global.console.error = (...args) => {
      const propTypeFailures = [/Warning:/] // e.g. [/Invalid prop/, /Failed prop type/]

      if (propTypeFailures.some(p => p.test(args[0]))) {
        throw new Error(args[0])
      }

      originalConsoleError(...args)
    }
  })
}

/**
 * Mix an match help from:
 * https://daveceddia.com/testing-for-proptypes-errors/
 * https://www.jackfranklin.co.uk/blog/failing-tests-on-react-proptypes/
 */
