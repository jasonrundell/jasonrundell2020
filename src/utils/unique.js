const onlyUnique = (value, index, self) => {
  return self.indexOf(value) === index
}

const generatePassword = ({
  passwordLength = 16,
  charset = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_-=~`',
}) => {
  let retVal = ''
  for (var i = 0, n = charset.length; i < passwordLength; ++i) {
    retVal += charset.charAt(Math.floor(Math.random() * n))
  }
  return retVal
}

export { onlyUnique, generatePassword }
