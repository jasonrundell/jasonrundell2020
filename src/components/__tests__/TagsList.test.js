import React from 'react'
import { unmountComponentAtNode } from 'react-dom'
import renderer from 'react-test-renderer'
import { catchWarnings } from '../../utils/test-utils'

import TagsList from '../Lists/TagsList'

catchWarnings()

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

const listData = ['one', 'two', 'three']

describe('TagsList component', () => {
  it('TagsList renders correctly', () => {
    const tree = renderer
      .create(
        <div>
          <TagsList list={listData} />
          <TagsList list={listData} orderBy="ASC" />
          <TagsList list={listData} orderBy="DESC" />
        </div>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
