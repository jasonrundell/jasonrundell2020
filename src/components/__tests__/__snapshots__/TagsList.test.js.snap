// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsList component TagsList renders correctly 1`] = `
<div>
  <ul
    className="root"
  >
    <li
      className="listItem"
    >
      <div>
        one
      </div>
    </li>
    <li
      className="listItem"
    >
      <div>
        two
      </div>
    </li>
    <li
      className="listItem"
    >
      <div>
        three
      </div>
    </li>
  </ul>
  <ul
    className="root"
  >
    <li
      className="listItem"
    >
      <div>
        one
      </div>
    </li>
    <li
      className="listItem"
    >
      <div>
        two
      </div>
    </li>
    <li
      className="listItem"
    >
      <div>
        three
      </div>
    </li>
  </ul>
  <ul
    className="root"
  >
    <li
      className="listItem"
    >
      <div>
        one
      </div>
    </li>
    <li
      className="listItem"
    >
      <div>
        two
      </div>
    </li>
    <li
      className="listItem"
    >
      <div>
        three
      </div>
    </li>
  </ul>
</div>
`;
